pkg/kubelet/cm/qos_container_manager_linux.go

```diff
@@ -159,7 +159,11 @@ func (m *qosContainerManagerImpl) setHugePagesUnbounded(cgroupConfig *CgroupConf
 }
 
 func (m *qosContainerManagerImpl) setHugePagesConfig(configs map[v1.PodQOSClass]*CgroupConfig) error {
-	for _, v := range configs {
+	for qosClass, v := range configs {
+		// the top guaranteed QOS class should respect the reservation for huge pages
+		if qosClass == v1.PodQOSGuaranteed {
+			continue
+		}
```

Comment:
This hugepages limit value is set to the correct values during `enforceNodeAllocatableCgroups`, but qos container manager is wrongly to refresh it to maximum.
https://github.com/kubernetes/kubernetes/blob/69b648a1d7074cbe004bf7adb3cdb17f01a4e9d8/pkg/kubelet/cm/node_container_manager_linux.go#L68-L114